bpf/lib/policy.h

```diff
@@ -228,18 +223,28 @@ __policy_can_access(const void *map, struct __ctx_buff *ctx, __u32 local_id,
 	if (policy) {
 		account(ctx, policy);
 		*match_type = POLICY_MATCH_ALL;
-		if (unlikely(policy->deny))
-			return DROP_POLICY_DENY;
-		return CTX_ACT_OK;
+		goto policy_check_entry;
 	}
 
-	if (ctx_load_meta(ctx, CB_POLICY))
+	/* TODO: Consider skipping policy lookup in this case? */
+	if (ctx_load_meta(ctx, CB_POLICY)) {
```

Comment:
I was simply thinking that this code block is not dependent on the preceding map lookups, so moving it to the beginning of the function would strictly reduce the number of map lookups (in the case `ctx_load_meta(ctx, CB_POLICY) != 0`), and not increase map lookups in any case, as `ctx_load_meta()` is not a map lookup?

What I did not consider before is that this would also skip denies due to explicitly hitting a deny policy. But then, the policy verdict is ignored when CT state is REPLY or RELATED, and indeed we do skip the whole policy lookup in that case in another commit in this series.